### PR TITLE
[Tables] - Sticky headers

### PR DESCRIPTION
**Pull request by dtailor90**
Headers should always stay visible, even when user scrolls the table.
Fixes issue #1481 

## Author checklist:
Changes address original issue? Y
Unit tests included and/or updated with changes? Y
Command line build passes? Y
Changes have been smoke-tested? Y

### CODE REVIEW

**Comment by charlesh88** [comment]
@akhenry @dtailor90  I will take on the horizontal scrollbar problem.

**Comment by charlesh88** [comment]
Oy. A pure CSS solution for this is currently eluding my feeble brain right now. The problem is that we need the body to independently scroll vertically (which it does now) but the headers must scroll horizontally with the body, when the body is in a horizontally overflowing situation. Putting a horizontal scroll on a wrapper around the header and body doesn't work, because the vertical scrollbar for the body gets hidden when we're horizontally overflowing, and is only visible when the user horizontally scrolls all the way to the right. 
Basically, this needs to work like a Google Spreadsheet. We used `<mct-scroll>` in the Timeline for a similar solution, that may need to be used here. @dtailor90 Let's have a look at this.

**Comment by dtailor90** [comment]
@akhenry - This is ready for re-review

File: platform/features/table/res/templates/mct-table.html [hunk 1]
```diff
@@ -43,14 +32,32 @@
                         width: columnWidths[$index] + 'px',
                         'max-width': columnWidths[$index] + 'px',
                     }">
-                    <div class="holder l-filter flex-elem grows">
-                        <input type="search"
+                    <input type="text"
                             ng-model="filters[header]"/>
-                        <a class="clear-icon clear-input icon-x-in-circle" ng-class="{show: filters[header]}" ng-click="filters[header] = undefined"></a>
```
Comment (larkin) [inline platform/features/table/res/templates/mct-table.html]: This was added to fix #1989 and should not be removed.  Need to restore this functionality.

**Comment by dtailor90** [comment]
@charlesh88 - i added the clear icon back, but there is a subtle style issue. The magnifier icon is rendering over the input text.

File: platform/features/table/res/templates/mct-table.html (hunk 1 above)
Comment (larkin) [reply platform/features/table/res/templates/mct-table.html]: Has been resolved.

**Comment by larkin** [comment]
# Reviewer Checklist

1. Changes appear to address issue? Y
2. Appropriate unit tests included? Y
3. Code style and in-line documentation are appropriate? Y
4. Commit messages meet standards? Y
